tt_metal/detail/reports/memory_reporter.cpp

```diff
@@ -17,6 +20,12 @@ using bank_to_statistics = std::map<uint32_t, allocator::Statistics>;
 
 std::atomic<bool> MemoryReporter::is_enabled_ = false;
 
+MemoryReporter::MemoryReporter() {
+    if (parse_env<bool>("TT_METAL_ENABLE_MEMORY_REPORTS", false)) {
```

Comment:
not a huge fan of env vars, do we need this because we have an api to generate memory reports already

Reply:
I suppose we don't need this, I just find it cumbersome to go find the test / edit the source just to add `tt_lib.device.EnableMemoryReports()` / `tt_lib.device.DumpMemoryReports()`.  It's nice to just flip a switch in your env.  I'm open to other ideas :)

I'm not a huge fan of envs either, but I think debugging / profiling / logging etc are reasonable exceptions to this.

Reply:
Im still in favour of the cumbersome approach (sorry 😬)

Reply:
Fair enough :)